test/TestInternalGrains/EchoTaskGrain.cs

```diff
@@ -51,9 +51,15 @@ public Task<string> EchoError(string data)
     }
 
     [StorageProvider(ProviderName = "MemoryStore")]
-    public class EchoTaskGrain : Grain<EchoTaskGrainState>, IEchoTaskGrain
+    internal class EchoTaskGrain : Grain<EchoTaskGrainState>, IEchoTaskGrain
     {
-        private  Logger logger;
+        private readonly IInternalGrainFactory internalGrainFactory;
+        private Logger logger;
+
+        public EchoTaskGrain(IInternalGrainFactory internalGrainFactory)
+        {
+            this.internalGrainFactory = internalGrainFactory;
```

Comment:
does this work? are we always using DI to create new grains? I thought GrainCreator had logic to only do it if the user used Startup to provide his own SP.
If it doesn't work, then consider just getting the IInternalGrainFactory from the property that has the SP. Not ideal, but back compat.